sdk/keyvault/keyvault-secrets/review/keyvault-secrets.review.api.md

```diff
@@ -0,0 +1,151 @@
+## API Report File for "@azure/keyvault-secrets"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import { HttpClient } from '@azure/core-http';
+import { HttpPipelineLogger } from '@azure/core-http';
+import * as msRest from '@azure/core-http';
+import { PagedAsyncIterableIterator } from '@azure/core-paging';
+import { PageSettings } from '@azure/core-paging';
+import { RequestOptionsBase } from '@azure/core-http';
+import { ServiceClientOptions } from '@azure/core-http';
+import { TokenCredential } from '@azure/core-http';
+
+// @public
+export interface DeletedSecret extends Secret {
+    readonly deletedDate?: Date;
+    recoveryId?: string;
+    readonly scheduledPurgeDate?: Date;
+}
+
+// @public
+export type DeletionRecoveryLevel = "Purgeable" | "Recoverable+Purgeable" | "Recoverable" | "Recoverable+ProtectedSubscription";
+
+// @public
+export interface GetSecretOptions {
+    requestOptions?: msRest.RequestOptionsBase;
+    version?: string;
+}
+
+// @public
+export interface GetSecretsOptions {
+    requestOptions?: msRest.RequestOptionsBase;
+}
+
+// @public
+export interface NewPipelineOptions {
+    // (undocumented)
+    HTTPClient?: HttpClient;
+    // (undocumented)
+    logger?: HttpPipelineLogger;
+    // (undocumented)
+    proxyOptions?: ProxyOptions;
+    // (undocumented)
+    retryOptions?: RetryOptions;
+    telemetry?: TelemetryOptions;
+}
+
+export { PagedAsyncIterableIterator }
+
+export { PageSettings }
+
+// @public (undocumented)
+export interface ParsedKeyVaultEntityIdentifier {
+    name: string;
+    vaultUrl: string;
+    version?: string;
+}
+
+// @public
+export interface ProxyOptions {
+    // (undocumented)
+    proxySettings?: string;
+}
+
+// @public
+export interface RetryOptions {
+    readonly maxRetryDelayInMs?: number;
+    readonly retryCount?: number;
+    readonly retryIntervalInMS?: number;
+}
+
+// @public
+export interface Secret {
+    properties: SecretProperties;
+    value?: string;
+}
+
+// @public
+export interface SecretProperties extends ParsedKeyVaultEntityIdentifier {
+    contentType?: string;
+    readonly created?: Date;
+    enabled?: boolean;
+    expires?: Date;
+    id?: string;
+    readonly keyId?: string;
+    readonly managed?: boolean;
+    notBefore?: Date;
+    readonly recoveryLevel?: DeletionRecoveryLevel;
+    tags?: {
+        [propertyName: string]: string;
+    };
+    readonly updated?: Date;
+}
+
+// @public
+export class SecretsClient {
+    constructor(url: string, credential: TokenCredential, pipelineOrOptions?: ServiceClientOptions | NewPipelineOptions);
+    backupSecret(secretName: string, options?: RequestOptionsBase): Promise<Uint8Array | undefined>;
```

Comment:
backup, delete, getDeleted, purge, recover, and restore here all use `RequestOptionsBase` directly as their options parameter. In `keyvault-keys`, these methods use a wrapper type `RequestOptions` (`{ requestOptions? : msRest.RequestOptionsBase }`). I think this package should do the same thing, both for consistency's sake and because it will allow us to add optional keys to the request options type without completely breaking the API signature.